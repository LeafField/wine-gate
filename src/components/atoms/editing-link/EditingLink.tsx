import React, { FC } from "react";
import Link from "next/link";

type Props = {
  wine_id: string;
};

const EditingLink: FC<Props> = ({ wine_id }) => {
  return (
    <Link
      className="block w-fit bg-blue/90 px-3 py-[0.375rem] text-white transition-colors duration-300 hover:bg-blue"
      href={`/editing/${wine_id}`}
      prefetch={false}
    >
      記事編集
    </Link>
  );
};

export default EditingLink;
