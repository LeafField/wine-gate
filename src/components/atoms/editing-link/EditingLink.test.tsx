import { screen, render } from "@testing-library/react";
import EditingLink from "./EditingLink";

describe("EditingLinkの単体テスト", () => {
  test("propsのwine_idに応じて正しいリンク先がaタグのhrefに設定される", () => {
    const { container } = render(<EditingLink wine_id="1234" />);
    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("href", "/editing/1234");
    expect(container).toMatchSnapshot();
  });
});
