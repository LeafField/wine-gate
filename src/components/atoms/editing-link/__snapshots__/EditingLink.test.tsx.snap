// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditingLinkの単体テスト propsのwine_idに応じて正しいリンク先がaタグのhrefに設定される 1`] = `
<div>
  <a
    class="block w-fit bg-blue/90 px-3 py-[0.375rem] text-white transition-colors duration-300 hover:bg-blue"
    href="/editing/1234"
  >
    記事編集
  </a>
</div>
`;
